### PR TITLE
Chore: Update more rules to use the new PassesTest method instead of Evaluate

### PR DESCRIPTION
**Pull request by RobGallo**
#### Describe the change

Adding the evaluation code to the rule's info will allow us to automatically generate documentation which provides the rule's expected error code.

### CODE REVIEW

File: src/Rules/Library/ButtonInvokeAndTogglePatterns.cs [hunk 1]
```diff
@@ -16,13 +16,14 @@ public ButtonInvokeAndTogglePatterns()
...
+            this.Info.ErrorCode = EvaluationCode.Error;
         }
 
-        public override EvaluationCode Evaluate(IA11yElement e)
+        public override bool PassesTest(IA11yElement e)
         {
-            var rule = Relationships.All(Patterns.Invoke, Patterns.Toggle);
+            var rule = ~(Relationships.All(Patterns.Invoke, Patterns.Toggle));
 
-            return rule.Matches(e) ? EvaluationCode.Error : EvaluationCode.Pass;
```
Comment (DaveTryon) [inline src/Rules/Library/ButtonInvokeAndTogglePatterns.cs:25]: Would it be any more efficient to leave rule unchanged, then negate the return value from Matches? That saves us creating a Condition object, and I think you've followed this pattern in previous changes

File: src/Rules/Library/ButtonToggleAndExpandeCollapsePatterns.cs [hunk 2]
```diff
@@ -16,13 +16,14 @@ public ButtonToggleAndExpandCollapsePatterns()
...
             this.Info.HowToFix = HowToFix.ButtonToggleAndExpandCollapsePatterns;
             this.Info.Standard = A11yCriteriaId.NameRoleValue;
+            this.Info.ErrorCode = EvaluationCode.Error;
         }
 
-        public override EvaluationCode Evaluate(IA11yElement e)
+        public override bool PassesTest(IA11yElement e)
         {
-            var rule = Relationships.All(Patterns.Toggle, Patterns.ExpandCollapse);
+            var rule = ~(Relationships.All(Patterns.Toggle, Patterns.ExpandCollapse));
```
Comment (DaveTryon) [inline src/Rules/Library/ButtonToggleAndExpandeCollapsePatterns.cs:24]: Would it be any more efficient to leave rule unchanged, then negate the return value from Matches? That saves us creating a Condition object, and I think you've followed this pattern in previous changes

**Review by DaveTryon — APPROVED** [review]
Approved with a couple of comments

File: src/Rules/Library/ButtonInvokeAndTogglePatterns.cs (hunk 1 above)
Comment (RobGallo) [reply src/Rules/Library/ButtonInvokeAndTogglePatterns.cs:25]: I did that originally, and then changed it to this way. Honestly, I'm not crazy about the '!' negation, even in the cases I've used it. I think it makes it harder to easily groc the intent of the method.

**Pull request merged**